### PR TITLE
rfc(ui): Add more frontend performance instrumentation

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
In order to get more accurate transaction.duration, this includes more instrumentation around the app (as a whole) and certain views to further break down render.

This bumps idleTimeout to deal with longer requests causing additional spans to be dropped, includes a profiler around app to catch render durations, and wraps some of the primary components used for performance/discover.

The `App` updates are only recorded above a threshold to not include too many spans inside the frontend transaction.

### Screenshots
Screen Shot 2020-09-09 at 10 32 04 AM
_**Showing app update and the GridEditable update that causes it**_
.

Screen Shot 2020-09-09 at 10 25 33 AM
**_Shows the critical path gap caused by rendering before the network call is made_**

### CODE REVIEW

**Comment by billyvg** [comment]
Just a note, `Profiler` is only available in non-production builds.

What do we gain with Profiler vs using `withProfiler` from our SDK?

**Comment by k-fish** [comment]
@billyvg yeah I know, it's possible to enable it though, iirc? The difference between `Profiler` and `withProfiler` is that our `withProfiler` doesn't actually include render duration, all updates are event only, so it doesn't give you an indication of render time, just that it did update. It's probably something we should into.

**Comment by markstory** [comment]
I like the possibility of being able to track component render times, and the frequency of re-renders in production. The re-renders can be useful in explaining gaps between networks requests or if there is a high volume of re-renders in a cascade.

**Comment by dashed** [comment]
@rhcarvalho Can you skim if this PR looks good from your perspective?

**Review by rhcarvalho — APPROVED** [review]
> @rhcarvalho Can you skim if this PR looks good from your perspective?

This looks fine to me.

Is there any feedback for things we can consider upstreaming into the SDK?

**Comment by k-fish** [comment]
I need to still check the effect of changing the production build to allow profiling, it may increase our bundle size too much for this change to be worth it.

@rhcarvalho I would say we might upstream the function provided to `Profiler` into `@sentry/react` (https://github.com/getsentry/sentry/pull/20687/files#diff-742e19adc0925f54bc0e135c79da2176R10) , but it has to come with instructions on how to make sure your react build allows profiling in production, and we should test this out first ourselves to make sure it's useful.

**Comment by dashed** [comment]
@k-fish you can make use of this pattern to have a production and development codepath https://github.com/getsentry/sentry/blob/59934f51c005e59c887569323cf1adf3fc12c467/src/sentry/static/sentry/app/bootstrap.tsx#L32

Development codepaths will be stripped out during minification due to being marked and identified as deadcode.

**Comment by k-fish** [comment]
Looks like including profiling in production is minimal, ~2kb. 

#### vendor.js
Screen Shot 2020-09-10 at 11 29 54 AM

#### specific changes
Screen Shot 2020-09-10 at 11 29 13 AM
Screen Shot 2020-09-10 at 11 29 33 AM

File: src/sentry/static/sentry/app/bootstrap.tsx [hunk 1]
```diff
@@ -61,6 +61,7 @@ function getSentryIntegrations(hasReplays: boolean = false) {
         Router.createRoutes(routes()),
         Router.match as any
       ),
+      idleTimeout: 2000,
```
Comment (dashed) [inline src/sentry/static/sentry/app/bootstrap.tsx:64]: This will be overridden by https://github.com/getsentry/sentry/pull/20738

Comment (k-fish) [reply src/sentry/static/sentry/app/bootstrap.tsx:64]: Just as a heads up Alberto, I think that could increase transaction times, as now whatever the users does within 5 seconds will extend the transaction.

**Pull request closed without merging**